extras/test/test_MicroNMEA.regressiontest

```diff
@@ -136,5 +147,38 @@ Latitude     54010000 (54.010000 deg)
 Longitude    -2785000 (-2.785000 deg)
 Alt. valid   true
 Altitude     42200 (42.200m)
+GeoHt valid  true
+GeoHt        49100 (49.100m)
+--------------
+# Message without height above the geoid
+$GPGGA,111309.00,5400.6000,N,00247.1000,W,1,16,00.7,042.2,M,,,,*15
+Sentence    $GPGGA,111309.00,5400.6000,N,00247.1000,W,1,16,00.7,042.2,M,,,,*15
+Valid        true
+TalkerID     P
+MessageID    GGA
+Time         11:13:09.00
+Num sat      16
+HDOP         0.7
+Latitude     54010000 (54.010000 deg)
+Longitude    -2785000 (-2.785000 deg)
+Alt. valid   true
+Altitude     42200 (42.200m)
+GeoHt valid  false
+--------------
+# Message with invalid height above the geoid
+$GPGGA,111309.00,5400.6000,N,00247.1000,W,1,16,00.7,042.2,M,invalid,,,*64
+Sentence    $GPGGA,111309.00,5400.6000,N,00247.1000,W,1,16,00.7,042.2,M,invalid,,,*64
+Valid        true
+TalkerID     P
+MessageID    GGA
+Time         11:13:09.00
+Num sat      16
+HDOP         0.7
+Latitude     54010000 (54.010000 deg)
+Longitude    -2785000 (-2.785000 deg)
+Alt. valid   true
+Altitude     42200 (42.200m)
+GeoHt valid  true
+GeoHt        0 (0.000m)
```

Comment:
This is obviously wrong, but I see no way of fixing it without completely changing the way parsing works: Right now, `parseFloat` will happily skip everything non-numerical and parse it as a `0`.

Reply:
I think the correct fix is to have `parseFloat()` take an additional parameter (a pointer) that indicates if its result is valid or not.

After a quick glance I think maybe set the extra parameter to `false` at the start of `parseFloat()`, and then set to `true`  in the `while(isdigit())` loop would probably do it. I'll see what I can do to add this; it might be a few days until I get time to properly implement and test something.

Reply:
See #17